site/styles/_variables.scss

```diff
@@ -4,36 +4,88 @@
 // These variables should be reasonably generic so that overrides are placed in main.scss.
 
 :root {
-  --primary-rgb: 0, 163, 255;
-  --secondary-rgb: 102, 199, 255;
-  --body-color-rgb: 255, 255, 255;
-  --body-bg-rgb: 15, 15, 15;
-  --body-bg-accent-rgb: 25, 25, 25; // darken(body-bg, 10%) OR lighten(body-bg, 10%)
-  --link-color-hover-rgb: 217, 217, 217; // darken(body-color, 15%)
-  --input-focus-border-color-rgb: 131, 210, 216; // lighten(primary, 25%)
-  --main-content-icon-hover-rgb: 207, 207, 207; // white .8 opacity
-
-  --primary: rgb(var(--primary-rgb));
-  --secondary: rgb(var(--secondary-rgb));
-  --body-color: rgb(var(--body-color-rgb));
-  --body-bg: rgb(var(--body-bg-rgb));
-  --body-bg-accent: rgb(var(--body-bg-accent-rgb));
-  --link-color: var(--body-color);
-  --link-color-hover: rgb(var(--link-color-hover-rgb));
-  --input-focus-border-color: rgb(var(--input-focus-border-color-rgb));
-  --main-content-icon-hover: rgb(var(--main-content-icon-hover-rgb));
+  /*#region BASE COLOURS */
+    // Don't put derivative colour variables in this region.
+    
+    --primary-rgb: 0, 163, 255;
+    --secondary-rgb: 102, 199, 255;
+    --body-color-rgb: 255, 255, 255;
+    --body-bg-rgb: 0, 0, 0;
+
+    // To calculate --body-color-hover-rgb, do:
+    //   1.  Convert var(--body-color-rgb) to hex.
+    //   2a. If dark themed, run it through the Sass darken function:  darken(hex, 20%).
+    //   2b. If light themed, run it through the Sass lighten function:  lighten(hex, 20%).
+    //   3.  Convert the result back to RGB.
+    //   TODO(jackh): figure out a non-Sass way of calculating this.
+    --body-color-hover-rgb: 204, 204, 204;
+
+    // To calculate --input-focus-border-color-rgb, do:
+    //   1.  Convert var(--body-bg) to hex.
+    //   2a. If dark themed, run it through the Sass lighten function:  lighten(hex, 10%).
+    //   2b. If light themed, run it through the Sass darken function:  darken(hex, 10%).
+    //   3.  Convert the result back to RGB.
+    //   TODO(jackh): figure out a non-Sass way of calculating this.
+    --body-bg-accent-rgb: 25, 25, 25;
+
+    // To calculate --input-focus-border-color-rgb, do:
+    //   1.  Convert var(--primary-rgb) to hex.
+    //   2.  Run it through the Sass lighten function:  lighten(hex, 25%).
+    //   3.  Convert the result back to RGB.
+    //   TODO(jackh): figure out a non-Sass way of calculating this.
+    --input-focus-border-color-rgb: 131, 210, 216;
+    
+    // To calculate --hover-background-rgb, do:
+    //   1.  Convert var(--body-bg-accent-rgb) to hex.
+    //   2a. If dark themed, run it through the Sass lighten function:  lighten(hex, 13%).
+    //   2b. If light themed, run it through the Sass darken function:  darken(hex, 13%).
+    //   3.  Convert the result back to RGB.
+    //   TODO(jackh): figure out a non-Sass way of calculating this.
```

Comment:
Might need to tidy up some comments

Reply:
Looks like ` TODO(jackh)` comments are still there at couple of places. Do you want to keep it there lol?

Reply:
Yeah they are TODOs...

Reply:
Looks bit weird if there are TODO comments with a name on the main branch as this is a public repo.

Reply:
There's nothing unusual about having TODOs and the name is so people know who to talk to about it if they need to.